src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,353 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace {
+// Zero'th index will be used for unit tests only.
+static constexpr size_t kActionsDelegateTableSize =
+#ifdef ZCL_USING_ACTIONS_CLUSTER_SERVER
+    MATTER_DM_ACTIONS_CLUSTER_SERVER_ENDPOINT_COUNT +
+#endif
+    CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT + 1;
+
+} // namespace
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Actions {
+
+// Last byte is reserved of '\0' terminator.
```

Comment:
Action names are allowed to be 128 UTF-8 bytes, with no reference to "terminators".  This bit here is not spec-compliant.

Furthermore, assuming no embedded `\0` in strings is a bad idea.  Our strings always know their lengths, and we should just keep track of that.

I'm stopping here.  Please re-request review when this is actually spec-compliant, and when we are not using C-style strings for spec strings.

Reply:
Got it. Used the string without a null terminator.